### PR TITLE
feat(server): add SSR test for renderHook

### LINKED ISSUE #605: Usage in close-to-real SSR environment?
Is there any way to test SSR functionality in environment close to real SSR?

I'm testing the hook that should fallback to noop during SSR, the hook declaration itself, as for now i'm checking SSR by testing `window` type: `typeof window !== 'undefined'`. Like this:
```ts
export const myAwesomeHook = typeof window === 'undefined' ? ()=>{} : ()=>{/*some real hook code*/}
```

When testing with jest in it's common env (jsdom) window object is obviously exists and there is no way to test that everything will be fine in SSR, since i gaij browser version of a hook.
If to switch environment to `node`, ssr tests failing with
```
    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/en/configuration#testenvironment-string.
    Consider using the "jsdom" test environment.

    ReferenceError: document is not defined

       7 |
    >  8 |     renderHook(() => myAwesomeHook());
         |     ^
       9 |

```

In this case renderHook improted from server: `import { renderHook } from '@testing-library/react-hooks/server';`

### PR DESCRIPTION
**Pull request by jsjoeio**
**What**:

This PR adds a new test based on the work done in #607 to fix #605. Partially related to https://github.com/testing-library/react-hooks-testing-library/issues/796

**Why**:

I noticed a test wasn't added when that bug was fixed. I decided to add a test so we could prevent regressions.

**How**:

A new test was added which runs in a Node environment in Jest. 

**Checklist**:

- [x] Documentation updated
- [x] Tests
- [x] Ready to be merged
      
- [x] Added myself to contributors table

### CODE REVIEW

**Comment by jsjoeio** [comment]
>        Warning: It looks like you called ReactTestUtils.act(...) in a non-browser environment. If you're using TestRenderer for your tests, you should call ReactTestRenderer.act(...) instead of ReactTestUtils.act(...).

I looked at `src/__tests__/autoDetectRenderer.test.ts` because it has a `reactTestRenderer` but it looks custom 🤔 Any tips on how to fix this?

**Comment by jsjoeio** [comment]
Hey @mpeyper - looks like the test is failing. Would love to get your insights to fix this 🙏

**Comment by mpeyper** [comment]
The test looks fine, I’ll see if I can get to the bottom of it.

File: src/server/pure.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ function createServerRenderer<TProps, TResult>(
 ) {
   let renderProps: TProps | undefined
   let container: HTMLDivElement | undefined
-  let serverOutput: string = ''
+  let serverOutput = ''
   const testHarness = createTestHarness(rendererProps, wrapper, false)
 
   return {
```
Comment (mpeyper) [inline src/server/pure.ts:19]: github won't let me comment on the actual line, but the error in the test is being caused by the `act` call on line 22.  For some reason, when you use `react-dom@16.9.0` this tried to use the document which throws an error when testing in the node environment, which doesn't occur in newer versions.

You can replicate this by running:

```sh
npm run install:react-16.9.0
npm test -- --no-watch src/__tests__/ssr.test.ts
```

In the context of server rendering, the `act` call is not really necessary here.  Dropping it out still passes all of our tests and allows your new test to pass with the older version:

```ts
    render(props?: TProps) {
      renderProps = props
      try {
        serverOutput = ReactDOMServer.renderToString(testHarness(props))
      } catch (e: unknown) {
        rendererProps.setError(e as Error)
      }
    },
```

Comment (jsjoeio) [reply src/server/pure.ts:19]: Wow, I don't think I would have figured that out on my own 😅 I'll test this out and make the necessary changes. Thanks a bunch!

**Comment by jsjoeio** [comment]
Well, that worked! Of course now Netlify is failing lol

```js
12:34:31 PM: error Error in "/opt/build/repo/node_modules/gatsby-plugin-mdx/gatsby-node.js": globalThis is not defined
```

**Comment by mpeyper** [comment]
Yeah, the MetLife build had been failing force while and I’m not sure how to fix it. I’ve mostly been ignoring it with maintenance on this library wrapping up soon anyway.

This all looks fine to me and will try to get it merged next time I’m at my computer.

**Comment by jsjoeio** [comment]
@mpeyper any updates on this by chance? 👀

**Comment by mpeyper** [comment]
Sorry for the delay, and thanks again for the effort.

**Comment by jsjoeio** [comment]
woohoo! thanls for letting me bug you and for the tips!

**Pull request merged**